IsraelHiking.Web/src/application/services/route-statistics.service.spec.ts

```diff
@@ -195,8 +195,8 @@ describe("RouteStatisticsService", () => {
 
         let statistics = service.getStatistics(routeData, null, null, null, false);
 
-        expect(statistics.gain).toBeCloseTo(59.9,1);
-        expect(statistics.loss).toBeCloseTo(-59.8,1);
+        expect(statistics.gain).toBeCloseTo(37.5,1);
```

Comment:
Shouldn't the gain and lose be almost the same for this case?
There's a roughly 60 meter ascent and descent if you "filter out" the 700 and -700.
Is this due to the interpolation?

Reply:
I looked into it - turns out that these outliers don't pass the filtering in lines 88-95:
```
routeStatistics.points = routeStatistics.points.filter(p => !isNaN(p.latlng.alt) && p.latlng.alt != null);
for (let pointIndex = routeStatistics.points.length - 1; pointIndex > 0; pointIndex--) {
    let prevPoint = routeStatistics.points[pointIndex - 1];
    let currentPoint = routeStatistics.points[pointIndex];
    if (currentPoint.coordinate[0] - prevPoint.coordinate[0] < 0.001) {
        routeStatistics.points.splice(pointIndex, 1);
    }
}
```
Even after I change the outliers just enough to pass the filtering but still be considered as outliers, such outliers will usually not pass the resampling.
This invoked me to think and realize that we should resample at a higher resolution (lower diff between samples) and increase the median window size. I checked a few combinations on my set of routes and indeed this direction gives better results (my ground truth is google maps total climb/descent, which can be retrieved at the maplorer website). I chose the best combination considering all of my routes and changed the code to use it.

Reply:
That's great progress! When looking at the test and the expected value it still isn't clear.
The gain and loss of this test should be close to 60 and should be similar (i.e gain =~ loss).
I would advise to change the test to show this better, even change the input to pass the "close coordinates filter", and get filtered out in the median filter.
I don't know how to do this exactly, but I think it's a good use case to create.

Reply:
Thanks for pushing towards greater code :)
This was caused by some weird behavior of the moving median package we used, so I implemented our own moving median code.
Also, I changed the outlier removal test route to be symmetrical so we can more easily understand it, and to have a plateau at 70, so this peak won't be flattened out by the moving median filter.